test/prompt/invocation_layer/test_sagemaker.py

```diff
@@ -0,0 +1,158 @@
+import os
+from unittest.mock import patch, MagicMock, Mock
+
+import pytest
+
+from haystack.nodes.prompt.invocation_layer import SageMakerInvocationLayer
+
+
+@pytest.mark.skipif(
+    not os.environ.get("TEST_SAGEMAKER_MODEL_ENDPOINT", None), reason="Skipping because SageMaker not configured"
+)
+@pytest.mark.integration
+def test_supports():
+    """
+    Test that supports returns True for valid SageMakerInvocationLayer
+    """
+    model_name_or_path = os.environ.get("TEST_SAGEMAKER_MODEL_ENDPOINT")
+    assert SageMakerInvocationLayer.supports(model_name_or_path=model_name_or_path)
+
+
+@pytest.mark.skipif(
+    not os.environ.get("TEST_SAGEMAKER_MODEL_ENDPOINT", None), reason="Skipping because SageMaker not configured"
+)
+@pytest.mark.integration
+def test_supports_not():
```

Comment:
Let's use a more detailed test name. Sth like `test_not_supports_invalid_profile` or similar. Same for the previous test named `test_supports`.